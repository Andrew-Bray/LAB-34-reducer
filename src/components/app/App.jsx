/* eslint-disable max-len */
import React, { useReducer } from 'react';
import reducer, { initialState } from '../../reducers/recordReducer';

function App() {
  const [state, dispatch] = useReducer(reducer, initialState);

  const handleChange = (val) => {
    dispatch({
      type: 'RECORD',
      payload: val
    });
  };

  const handleClick = ({ target }) => {
    dispatch({
      type: target.id,
    });
  };

  return (
    <>
      <button 
        data-testid="undo-button" 
        id="UNDO" 
        onClick={handleClick}>
          undo
      </button>
      <button 
        data-testid="redo-button" 
        id="REDO"
        onClick={handleClick}>
          redo
      </button>
      <input 
        data-testid="color-picker"
        id="RECORD"
        type="color" 
        value={state.current} 
        onChange={({ target }) => handleChange(target.value)} />
      <div data-testid="my-color-div" style={{ backgroundColor: state.current, width: '10rem', height: '10rem' }}></div>
    </>
  );
}

export default App;
