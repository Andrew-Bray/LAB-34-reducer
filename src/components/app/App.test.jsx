import React from 'react';
import { render, cleanup, screen, fireEvent } from '@testing-library/react';
import App from './App';

describe('useRecord function', () => {
  it ('changes the background color', async() => {
    render(<App />);
    const colorInput = screen.getByTestId('color-picker');
    fireEvent.change(colorInput, {
      target: {
        value: '#00FF00'
      }
    });
    const div = await screen.getByTestId('my-color-div');
    expect(div).toHaveStyle({
      backgroundColor: '#00FF00'
    });
  });

  it('tests the undo button', async() => {
    render(<App />);
    const undo = screen.getByTestId('undo-button');
    const colorInput = screen.getByTestId('color-picker');
    fireEvent.change(colorInput, {
      target: {
        value: '#00FF00'
      }
    });  
    fireEvent.click(undo);
    
    const div = await screen.findByTestId('my-color-div');
    expect(div).toHaveStyle({
      backgroundColor: '#FF0000'
    });
  });

  it('tests the redo button', async() => {
    render(<App />);
    const undo = screen.getByTestId('undo-button');
    const redo = screen.getByTestId('redo-button');

    const colorInput = screen.getByTestId('color-picker');
    fireEvent.change(colorInput, {
      target: {
        value: '#00FF00'
      }
    });  
    fireEvent.click(undo);
    fireEvent.click(redo);
    
    const div = await screen.findByTestId('my-color-div');
    expect(div).toHaveStyle({
      backgroundColor: '#00FF00'
    });
  });





});

describe('App component', () => {
  afterEach(() => cleanup());
  it('renders App', () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });
});
