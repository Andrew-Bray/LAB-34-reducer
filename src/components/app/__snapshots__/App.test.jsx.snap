// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<DocumentFragment>
  <button
    data-testid="undo-button"
    id="UNDO"
  >
    undo
  </button>
  <button
    data-testid="redo-button"
    id="REDO"
  >
    redo
  </button>
  <input
    data-testid="color-picker"
    id="RECORD"
    type="color"
    value="#FF0000"
  />
  <div
    data-testid="my-color-div"
    style="background-color: rgb(255, 0, 0); width: 10rem; height: 10rem;"
  />
</DocumentFragment>
`;
